### PR TITLE
Update and Delete animal databases

### PR DESCRIPTION
**Pull request by MICHAELKITH**
### In this Milestone:

- [ ] Added another column, 'species' to table.

- [ ] Used required transactional queries on the new data.

- [ ] Used aggregate queries on the new data.

- [ ] Added new data to table.

image

image

image

image

image

image

image

image

image

image

### CODE REVIEW

File: data.sql [hunk 1]
```diff
@@ -1,9 +1,12 @@
-/* Populate database with sample data. */
-
-/* Populate database
-WITH sample data. */
-
 INSERT INTO animals VALUES (1, 'Agumon', '02/03/2020', 0, 'T', 10.23);
 INSERT INTO animals VALUES (2, 'Gabumon', '09/15/2018', 2, 'T', 8);
 INSERT INTO animals VALUES (3, 'Pikachu', '01/07/2021', 1, 'F', 15.04);
 INSERT INTO animals VALUES (4, 'Devimon', '05/12/2017', 5, 'T', 11);
+ALTER TABLE animals add column species char(30);
```
Comment (Zuhaib042) [inline data.sql]: - [ ] Kindly consider removing this line from here and adding this line to schema.sql file because everything that creates or change the tables structure should be saved in the schema.sql file.

File: schema.sql [hunk 2]
```diff
@@ -1,3 +1,5 @@
 /* Database schema to keep the structure of entire database. */
 
-CREATE TABLE animals( id int GENERATED BY DEFAULT AS Identity, name char(50), date_of_birth date, escape_attempts int, neutered boolean, weight_kg decimal );
\ No newline at end of file
+/* Database schema to keep the structure of entire database. */
+
+CREATE TABLE animals( id int GENERATED BY DEFAULT AS Identity, name char(50), date_of_birth date, escape_attempts int, neutered boolean, weight_kg decimal, species char(30) );
```
Comment (Zuhaib042) [inline schema.sql]: - [ ] You can undo this change and keep the animals table as it was before and add `ALTER TABLE animals add column species char(30);` here instead. This is just a suggestion for organizing your schema.sql file.

File: queries.sql [hunk 3]
```diff
@@ -32,4 +32,51 @@ WHERE name != 'Gabumon';
...
+FROM animals; rollback;
+
+SELECT  *
+FROM animals; begin; delete
+FROM animals
+WHERE date_of_birth > '01/01/2022'; savepoint neg; update animals
+
+SET weight_kg = (weight_kg * -1); rollback to neg; update animals
+SET weight_kg = (weight_kg * -1)
+WHERE weight_kg < 0; release savepoint neg; commit;
```
Comment (Zuhaib042) [inline queries.sql]: - [ ] Kindly consider structuring your queries where `BEGIN; UPDATE; COMMIT;` each is placed on a separate line so that the queries look organized and clean.
- [ ] You have not implemented the following part of the requirement and in result the `species` column is null in your table for all rows, so consider fixing it.
- Inside a transaction:
     - Update the animals table by setting the species column to digimon for all animals that have a name ending in 
     mon.
     - Update the animals table by setting the species column to pokemon for all animals that don't have species 
     already set.
     - Commit the transaction.
     - Verify that change was made and persists after commit.
- [ ] Kindly consider implementing the requirement one by one and also check where it asks you to check if some specific change in the table persist or not. and also take screenshot where it says to take a screen shot so that the reviewer can see the result of specific query.

File: queries.sql [hunk 4]
```diff
@@ -32,4 +32,51 @@ WHERE name != 'Gabumon';
...
+GROUP BY  neutered
+
+SELECT  species, MAX(weight_kg) AS max_weight, MIN(weight_kg) AS min_weight
+FROM animals
+GROUP BY  species
+SELECT  species,AVG(escape_attempts) AS avg_escapes
+FROM animals
+WHERE extract('year'
+FROM date_of_birth) BETWEEN 1990 AND 2000
+GROUP BY  species
```
Comment (Zuhaib042) [inline queries.sql]: - [ ] Kindly consider running these two queries after fixing the bug in species.`Your species should be either `Digimon where animal name ends with 'mon' or Pokemon where animal name doesn't end with 'mon'`.
After fixing the bug kindly consider running these two queries, taking screenshots and uploading those screenshots in the PR.

**Review by Zuhaib042 — CHANGES_REQUESTED** [review]
Hi @MICHAELKITH 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!


## Highlights
✅ Descriptive PR
✅ Correct Gitflow

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by KayLemba — APPROVED** [review]
[[# STATUS: APPROVED :green_circle: :clap: 

Hello @MICHAELKITH        , 
Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉  Keep up the good work 🥂 

HIGHLIGHTS:

- [x]  Your repo is professional  😎 
- [x] Descriptive P.R 💯 
- [x] linters are passing
- [x] Professional README 💯
- [x] Database running as intended  🥇  
 










 








Now you can merge this branch and move on :1st_place_medal: 
Happy Coding cheers 🥇 💯 



well-1

Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better.

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._]]
